### PR TITLE
chore: updating old button and removing deprecated disabled colors

### LINKED ISSUE #24962: chore: remove all instances of primary-disabled color
### What is this about?

In preparation for upgrading the `@metamask/design-tokens` library to version 4, we need to remove all instances of the deprecated `primary-disabled`. This color has been removed in v4 and its presence will cause breakages in the application.

[Design tokens migration v3 => v4 documentation](https://github.com/MetaMask/design-tokens/blob/main/MIGRATION.md#from-version-300-to-400)

### Scenario

_No response_

### Design

_No response_

### Technical Details

- Search the codebase for all instances of `primary-disabled`.
- Replace `primary-disabled` with an appropriate alternative from the updated design tokens.
- Ensure all components using this color are updated accordingly and tested for visual consistency.
- Update any documentation or style guides referencing `primary-disabled`.

### Threat Modeling Framework

_No response_

### Acceptance Criteria

- All instances of `primary-disabled` are removed from the codebase.
- The application functions correctly without any references to `primary-disabled`.
- Visual appearance of components previously using `primary-disabled` is consistent with design specifications.
- No breaking changes introduced in the application due to this update.

### Stakeholder review needed before the work gets merged

- [X] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This pull request removes the primary-disabled and error-disabled colors in preparation for upgrading to [design token v4](https://github.com/MetaMask/design-tokens/blob/main/MIGRATION.md#from-version-300-to-400). The updates include modifications to the old button component, and the removal of these colors from styles and helpers.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24970?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24962 https://github.com/MetaMask/metamask-extension/issues/24963

## **Manual testing steps**

1. Go to the storybook build of this PR
2. Go to the old button component and check that it works as expected for all types in disabled state
3. Search for deprecated instances of primary-disabled and error-disabled to ensure none remain

### **Before**

https://github.com/MetaMask/metamask-extension/assets/8112138/02b40217-8f5e-430c-9dab-bf637fd891d3

### **After**

https://github.com/MetaMask/metamask-extension/assets/8112138/e8ee564a-64f6-4fb1-8c1e-06ba1e2ab300

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: .storybook/3.COLORS.stories.mdx [hunk 1]
```diff
@@ -129,8 +131,8 @@ Another level of abstraction is component tier colors that you can define at the
   --color-text-primary: var(--color-primary-inverse);
   --color-border-primary: var(--color-primary-default);
 
-  --color-background-primary-hover: var(--color-primary-alternative);
-  --color-border-primary-hover: var(--color-primary-alternative);
+  --color-background-primary-hover: var(--color-primary-default-hover);
+  --color-border-primary-hover: var(--color-primary-default-hover);
 
   .btn-primary {
     background-color: var(--color-background-primary);
```
Comment (georgewrmarshall) [inline .storybook/3.COLORS.stories.mdx:138]: Updating color documentation

### Before

### After

https://github.com/MetaMask/metamask-extension/assets/8112138/68862453-0d5d-4b9e-ae87-52d18161215c

File: ui/components/ui/button/buttons.scss [hunk 2]
```diff
@@ -20,7 +20,7 @@
   &--disabled,
   &[disabled] {
     cursor: auto;
-    opacity: 0.5;
+    opacity: var(--opacity-disabled);
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:23]: Uses the global opacity token which = `0.5`

https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/component-library/component-library-components.scss#L10

File: ui/components/ui/button/buttons.scss [hunk 3]
```diff
@@ -33,23 +33,18 @@
...
+    background: var(--color-primary-default);
     border-color: var(--color-primary-default);
   }
-
-  &--disabled,
-  &[disabled] {
-    opacity: 1;
-    color: var(--color-primary-muted);
-  }
 }
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:48]: Secondary type still works as expected

### Before



### After

https://github.com/MetaMask/metamask-extension/assets/8112138/af85c3fa-a216-48ea-a6cd-f633de5d4931

File: ui/components/ui/button/buttons.scss [hunk 4]
```diff
@@ -86,12 +81,6 @@
     background: var(--color-error-muted);
     border-color: var(--color-error-default);
   }
-
-  &--disabled,
-  &[disabled] {
-    opacity: 1;
-    color: var(--color-error-disabled);
-  }
 }
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:84]: Warning disabled works as expected checking local branch against develop branch GitHub storybook


https://github.com/MetaMask/metamask-extension/assets/8112138/bb18c0af-7272-40df-bca5-6566a3a22511

File: ui/css/design-system/_colors.scss [hunk 5]
```diff
@@ -16,12 +16,10 @@ $color-map: (
   'primary-alternative': --color-primary-alternative,
   'primary-muted': --color-primary-muted,
   'primary-inverse': --color-primary-inverse,
-  'primary-disabled': --color-primary-disabled,
```
Comment (georgewrmarshall) [inline ui/css/design-system/_colors.scss:19]: No instances of primary-disabled exists

Screenshot 2024-05-31 at 4 28 46 PM

File: ui/css/design-system/_colors.scss [hunk 6]
```diff
@@ -16,12 +16,10 @@ $color-map: (
...
   'error-default': --color-error-default,
   'error-alternative': --color-error-alternative,
   'error-muted': --color-error-muted,
   'error-inverse': --color-error-inverse,
-  'error-disabled': --color-error-disabled,
```
Comment (georgewrmarshall) [inline ui/css/design-system/_colors.scss:24]: No instances of error-disabled exist in code base

Screenshot 2024-05-31 at 4 29 08 PM

File: ui/helpers/constants/design-system.ts [hunk 7]
```diff
@@ -22,12 +22,10 @@ export enum Color {
   primaryAlternative = 'primary-alternative',
   primaryMuted = 'primary-muted',
   primaryInverse = 'primary-inverse',
-  primaryDisabled = 'primary-disabled',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:25]: No instances of `primaryDisabled` used in codebase

Screenshot 2024-05-31 at 4 29 31 PM

File: ui/helpers/constants/design-system.ts [hunk 8]
```diff
@@ -22,12 +22,10 @@ export enum Color {
...
   errorDefault = 'error-default',
   errorAlternative = 'error-alternative',
   errorMuted = 'error-muted',
   errorInverse = 'error-inverse',
-  errorDisabled = 'error-disabled',
```
Comment (georgewrmarshall) [inline ui/helpers/constants/design-system.ts:30]: no instances of `errorDisabled` used in code base

Screenshot 2024-05-31 at 4 29 23 PM

File: ui/components/ui/button/buttons.scss (hunk 3 above)
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:48]: Checking secondary type button local build against develop github storybook works as expected


https://github.com/MetaMask/metamask-extension/assets/8112138/dc49ef1f-d58c-4d1e-bea5-f852e71df6b2

File: ui/components/ui/button/buttons.scss [hunk 9]
```diff
@@ -134,8 +116,7 @@
 
   &--disabled,
   &[disabled] {
-    opacity: 1;
-    color: var(--color-text-muted);
+    opacity: var(--opacity-disabled);
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:119]: Button default adheres to design system standards and works as expected when comparing against develop github storybook


https://github.com/MetaMask/metamask-extension/assets/8112138/e952318e-2961-42ca-8961-77cb0f865b00

File: ui/components/ui/button/buttons.scss [hunk 10]
```diff
@@ -109,13 +98,6 @@
     background: var(--color-error-alternative0);
     border-color: var(--color-error-alternative);
   }
-
-  &--disabled,
-  &[disabled] {
-    opacity: 1;
-    border-color: var(--color-error-disabled);
-    background-color: var(--color-error-disabled);
-  }
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:118]: Button danger primary works as expected checking local branch storybook against develop github storybook 


https://github.com/MetaMask/metamask-extension/assets/8112138/d9f84153-1aa6-49fb-a5ee-f0bbb4637c24

File: ui/components/ui/button/buttons.scss [hunk 11]
```diff
@@ -207,7 +180,7 @@
 button[disabled],
 input[type="submit"][disabled] {
   cursor: not-allowed;
-  opacity: 0.5;
+  opacity: var(--opacity-disabled);
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:183]: var(--opacity-disabled) = `0.5`

https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/component-library/component-library-components.scss#L10

File: ui/components/ui/button/buttons.scss [hunk 12]
```diff
@@ -179,9 +154,7 @@
   &--disabled,
   &[disabled] {
     cursor: auto;
-    opacity: 1;
     pointer-events: none;
-    color: var(--color-primary-disabled);
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:184]: Button link works as expected when comparing local against develop github storybook


https://github.com/MetaMask/metamask-extension/assets/8112138/ef302ab4-ba32-405d-8290-297875d9f17a

File: ui/components/ui/button/buttons.scss [hunk 13]
```diff
@@ -226,12 +199,6 @@ input[type="submit"][disabled] {
       box-shadow: var(--shadow-size-sm) var(--color-primary-shadow);
     }
 
-    &--disabled,
-    &[disabled] {
-      border-color: var(--color-primary-disabled);
-      color: var(--color-primary-disabled);
-    }
-
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:234]: duplicate CSS

File: ui/components/ui/button/buttons.scss [hunk 14]
```diff
@@ -244,12 +211,6 @@ input[type="submit"][disabled] {
       box-shadow: var(--shadow-size-sm) var(--color-shadow-default);
     }
 
-    &--disabled,
-    &[disabled] {
-      border-color: var(--color-border-muted);
-      color: var(--color-text-muted);
-    }
-
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:252]: duplicate CSS

File: ui/components/ui/button/buttons.scss [hunk 15]
```diff
@@ -262,12 +223,6 @@ input[type="submit"][disabled] {
       box-shadow: var(--shadow-size-sm) var(--color-error-shadow);
     }
 
-    &--disabled,
-    &[disabled] {
-      border-color: var(--color-error-disabled);
-      color: var(--color-error-disabled);
-    }
-
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:270]: duplicate CSS

File: ui/components/ui/button/buttons.scss [hunk 16]
```diff
@@ -298,11 +253,6 @@ input[type="submit"][disabled] {
       box-shadow: var(--shadow-size-sm) var(--color-primary-shadow);
     }
 
-    &--disabled,
-    &[disabled] {
-      background-color: var(--color-primary-disabled);
-    }
-
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:305]: duplicate CSS

File: ui/components/ui/button/buttons.scss [hunk 17]
```diff
@@ -315,11 +265,6 @@ input[type="submit"][disabled] {
       box-shadow: var(--shadow-size-sm) var(--color-error-shadow);
     }
 
-    &--disabled,
-    &[disabled] {
-      background-color: var(--color-error-disabled);
-    }
-
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:322]: duplicate CSS

File: ui/components/ui/button/buttons.scss [hunk 18]
```diff
@@ -346,8 +291,6 @@ input[type="submit"][disabled] {
   &--disabled,
   &[disabled] {
     cursor: auto;
-    opacity: 1;
     pointer-events: none;
-    color: var(--color-primary-disabled);
```
Comment (georgewrmarshall) [inline ui/components/ui/button/buttons.scss:351]: duplicate CSS

**Review by brianacnguyen — APPROVED** [review]

**Review by sahar-fehri — APPROVED** [review]

**Pull request merged**